src/libs/tmdk/tests/deploy.test.ts

```diff
@@ -1,26 +1,265 @@
+    });
+    fs.readFileSync = jest.fn((path: [string]) => {
+      if (path.includes("tmdk")) {
+        const tmdk1Ct = JSON.parse(JSON.stringify(emptyTmdk));
+        tmdk1Ct["component_types"] = ["componentType1.json"];
+        return JSON.stringify(tmdk1Ct, null, 4);
+      } else if (path.includes("componentType1.json")) {
+        return JSON.stringify(componentType1Definition, null, 4);
+      } else {
+        // covers entities.json
+        return JSON.stringify([], null, 4);
+      }
+    });
+    twinmakerMock.on(GetWorkspaceCommand).resolves({});
+    twinmakerMock
+      .on(GetComponentTypeCommand)
+      .rejectsOnce(
+        new ResourceNotFoundException({ $metadata: {}, message: "" })
+      )
+      .resolves({});
+
+    const argv2 = {
+      _: ["init"],
+      $0: "tmdk_local",
+      region: "us-east-1",
+      "workspace-id": workspaceId,
+      dir: fakeTmdkDir,
+    } as Arguments<Options>;
+    expect(await handler(argv2)).toBe(0);
+    expect(twinmakerMock.commandCalls(CreateComponentTypeCommand).length).toBe(
+      1
+    );
+    expect(
+      twinmakerMock.commandCalls(CreateComponentTypeCommand)[0].args[0].input
+    ).toStrictEqual({
+      workspaceId: workspaceId,
+      ...componentType1Definition,
+    });
+    expect(twinmakerMock.commandCalls(CreateEntityCommand).length).toBe(0);
+    expect(twinmakerMock.commandCalls(CreateSceneCommand).length).toBe(0);
+  });
+
+  test("deploy_given1SceneAnd1Model_expectSuccess", async () => {
+    fs.existsSync = jest.fn(() => {
+      return true;
+    });
+    fs.readFileSync = jest.fn((path: [string]) => {
+      if (path.includes("tmdk")) {
+        const tmdk1SceneAndModel = JSON.parse(JSON.stringify(emptyTmdk));
+        tmdk1SceneAndModel["scenes"] = ["scene1.json"];
+        tmdk1SceneAndModel["models"] = ["model1.glb"];
+        return JSON.stringify(tmdk1SceneAndModel, null, 4);
+      } else {
+        // covers entities.json
+        return JSON.stringify([], null, 4);
+      }
+    });
+    (fsPromises.readFile as jest.Mock).mockImplementation(async (path) => {
+      if (path.toString().includes("scene1")) {
+        return Promise.resolve(JSON.stringify(scene1, null, 4));
+      } else {
+        return Promise.resolve("empty readFile result");
+      }
+    });
+    twinmakerMock.on(GetWorkspaceCommand).resolves({
+      s3Location: s3BucketArn,
+    });
+
+    const argv2 = {
+      _: ["init"],
+      $0: "tmdk_local",
+      region: "us-east-1",
+      "workspace-id": workspaceId,
+      dir: fakeTmdkDir,
     } as Arguments<Options>;
-    await expect(handler(argv2)).rejects.toThrow(Error);
-    expect(mockExit).toHaveBeenCalledWith(1);
+    expect(await handler(argv2)).toBe(0);
+    expect(twinmakerMock.commandCalls(CreateComponentTypeCommand).length).toBe(
+      0
+    );
+    expect(twinmakerMock.commandCalls(CreateEntityCommand).length).toBe(0);
+    expect(twinmakerMock.commandCalls(CreateSceneCommand).length).toBe(1);
+    expect(
```

Comment:
there's a much cleaner way to assert on function called, using matchers such as `toHaveBeenCAlled() or `toHaveBeenLastCalledWith()`: https://jestjs.io/docs/mock-functions#custom-matchers

Reply:
I don't think this works for the aws mocking as I used the aws-sdk-mock package, not jest. I could not get the jest mock for s3 to work and for twinmaker it was unwieldy. Would be happy to discuss offline